tests/test_database.nim

```diff
@@ -10,35 +13,38 @@ proc checkValues(db: Database, vals: seq[seq[byte]]): bool =
 
     return true
 
-suite "Database Test Suite":
-
-    var db: Database
-
-    setup:
-        var result = database.open("/tmp")
-        check:
-            result.isOk 
-
-        db = result.value
-
-    teardown:
-        removeFile("/tmp/minima.db")
-
-    test "can set and get key":
-        let key = @[byte 1, 2, 3, 4]
-        let value = @[byte 4, 3, 2, 1]
+proc createEncryptedDatabase(): Database =
+    var res = database.open("/tmp", "password".toAESKey)
+    check:
+        res.isOk 
+    
+    return res.value
+
+proc createDatabase(): Database =
+    var res = database.open("/tmp")
+    check:
+        res.isOk 
+    
+    return res.value
+
+proc testSetAndGet(fn: InitProc) =
+    let key = @[byte 1, 2, 3, 4]
+    let value = @[byte 4, 3, 2, 1]
         
-        var setResult = db.set(key, value)
-        check:
-            setResult.isOk
+    let db = fn()
+    var setResult = db.set(key, value)
+    check:
+        setResult.isOk
 
-        var getResult = db.get(key)
+    var getResult = db.get(key)
```

Comment:
It seems unclear why `setResult`/`getResult` are `var`, since they're never mutated.